Update documentation for delete_entries and select_entries processors (#11476)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/delete-entries.md

 
# Delete entries processor

The `delete_entries` processor removes entries, such as key-value pairs, from an event. Use the `with_keys` field to specify
the exact keys to delete. To delete keys that match a regular expression pattern, use the `with_keys_regex` field. You can
prevent deletion of specific events when using regular expressions by configuring the `exclude_from_regex` field.

> To use both `with_keys` and `with_keys_regex` in the same configuration, define them under the `entries` field.

Reviewer: I think this is a little confusing. Do you mean to say that the pipeline author would need to create two different `entries`?

:--- |:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
| `with_keys` | No       | An array that specifies the keys of the entries to delete.                                                                                                                        |
| `with_keys_regex` | No       | An array of regular expression (regex) patterns used to match the keys of entries to delete.                                                                                      |
| `exclude_from_delete` | No | A set of entries to exclude from deletion when using the `with_keys_regex` configuration.                                                                                         |
| `entries` | No | A list of entries to delete from the event.                                                                                                                                       |

> | `delete_when` | No | Specifies under what conditions the deletion should be performed. (Example: <i>value= "/some_key == null"</i> - Deletion runs only if <i>some_key</i> is null or does not exist.) |

Reviewer 2: ```suggestion
| `delete_when` | No | Defines the condition under which the deletion is performed. For example, `value="/some_key == null"` deletes the key only if `/some_key` is null or does not exist. |
```

> |`delete_from_element_when` | No | Specifies the condition needed to delete the key from each element of a list when using `iterate_on`                                                                              |

Reviewer 2: @kennedy-onyia Thanks for clarifying - updated both suggestions.

pipeline:
  source:
    ...
  ....
  processor:
    - delete_entries:

>         entries:

Reviewer: I think we should have a completely different example for this rather than sharing the single example.

Reviewer 2: Thanks! Moving on to editorial review then.

When you run the `delete_entries` processor, it parses the message into the following output:

```json
{"message2": "goodbye", "message3": "test"}
```

### Additional configuration examples

#### Using `with_keys_regex` field

> To get started, create the following `pipeline.yaml` file:

Reviewer 2: ```suggestion
First, create the following `pipeline.yaml` file:

```
